### PR TITLE
Print some basic configuration info on startup

### LINKED ISSUE #665: When starting up in a terminal, step-ca should print some useful information
When started from the terminal, and when the logger format is not JSON, `step-ca` should always print:

- The root fingerprint & CA endpoints
- Configuration paths in use
- Link to docs (& Discord?)
- Client bootstrapping command?

### PR DESCRIPTION
**Pull request by tashian**
Fixes #665.

This still needs:
- [x] Print the `--context`
- [x] Passing `--quiet` should quiet the startup output
- [x] Print all CA endpoints instead of just "listening on :4443"

### CODE REVIEW

File: authority/authority.go [hunk 1]
```diff
@@ -313,6 +311,7 @@ func (a *Authority) init() error {
 	for _, crt := range a.rootX509Certs {
 		sum := sha256.Sum256(crt.Raw)
 		a.certificates.Store(hex.EncodeToString(sum[:]), crt)
+		log.Printf("X.509 Root Fingerprint: %s", hex.EncodeToString(sum[:]))
```
Comment (hslatman) [inline authority/authority.go:314]: I like this change

File: authority/authority.go [hunk 2]
```diff
@@ -541,6 +540,13 @@ func (a *Authority) init() error {
 		a.templates.Data["Step"] = tmplVars
 	}
 
+	if tmplVars.SSH.HostKey != nil {
+		log.Printf("SSH Host CA Key: %s\n", ssh.MarshalAuthorizedKey(tmplVars.SSH.HostKey))
+	}
+	if tmplVars.SSH.HostKey != nil {
```
Comment (hslatman) [inline authority/authority.go:546]:
```suggestion
	if tmplVars.SSH.UserKey != nil {
```

**Review by hslatman — COMMENTED** [review]
Nice starting point 🙂 

We should check that the new output works well with the product too. Since the `Authority.New` and thus `Authority.init` happen also in the product, we'll get the log output at startup there too. We might need to `--quiet` the entire startup info by default there.

One small nit is the fact that user input (e.g. passwords) is interleaved with output from the CA:

```console
badger 2022/03/22 09:06:32 INFO: All 1 tables opened in 1ms
badger 2022/03/22 09:06:32 INFO: Replaying file id: 0 at offset: 829498
badger 2022/03/22 09:06:32 INFO: Replay took: 2.625µs
Please enter the password to decrypt /Users/herman/.step/authorities/ca.localhost/secrets/intermediate_ca_key:
2022/03/22 09:06:33 X.509 Root Fingerprint: a09625e68c5d80640b4ce78fe4134c33e38783bc19e37b3e6b0d2fc843db4510
Please enter the password to decrypt /Users/herman/.step/authorities/ca.localhost/secrets/host.key:
Please enter the password to decrypt /Users/herman/.step/authorities/ca.localhost/secrets/user.key:
2022/03/22 09:06:36 SSH Host CA Key: ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBAX25QRTUuxH7TLwSLSOR3Zxb/XVBikuAcT1kFwnFMuR/aT0yA97bL6MgxXhlGXyAPIKlaSTWbpTimeVVDqJeRo=

2022/03/22 09:06:36 SSH User CA Key: ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYAAAAIbmlzdHAyNTYAAABBBOFm9A5eIsW9JqHN++F9FRzQZ4R7FlT8sFPLOojR6mNgyG4mimtT5asJV9dkMO4PuRdcBkH8ADSdBqciJpnrZO4=

2022/03/22 09:06:36 Documentation: https://u.step.sm/docs/ca
2022/03/22 09:06:36 Community Discord: https://u.step.sm/discord
2022/03/22 09:06:36 Config File: /Users/herman/.step/authorities/ca.localhost/config/ca.json
2022/03/22 09:06:36 Serving HTTPS on 127.0.0.1:7443 ...
```

It may be nicer to print all of the info at the end of initialization at once.

**Comment by tashian** [comment]
Thanks @hslatman. I'd love to put all of the startup output into one function in `ca/ca.go` or maybe `server/server.go`, but I'm not sure how to do that in terms of the variable scopes. In order to print this stuff, I need

- To construct the URLs that can be used to access the CA, I need:
  - The CA config's `dnsNames` (these could potentially be extracted from the leaf cert in `srv.TLSConfig`)
  - And the server port number (from `srv.Addr`, or possibly from the CA config)
- The root cert & SSH keys from the `Authority`
- The config file name from the CA options
- The current context (not sure where this comes from)

It's a start, but I'm a bit blocked on this right now.

**Comment by hslatman** [comment]
I think it's good to not start exposing the properties themselves directly from the `Authority`, but to expose a function that returns the info in a struct instead. Something like `Authority.GetInfo() AuthorityInfo`.

File: ca/ca.go [hunk 3]
```diff
@@ -288,6 +298,24 @@ func (ca *CA) Run() error {
 	var wg sync.WaitGroup
 	errs := make(chan error, 1)
 
+	if !ca.opts.quiet {
+		authorityInfo := ca.auth.GetInfo()
+		log.Printf("Address: %s", ca.config.Address)
+		for _, crt := range authorityInfo.RootX509Certs {
+			sum := sha256.Sum256(crt.Raw)
+			log.Printf("X.509 Root Fingerprint: %s", hex.EncodeToString(sum[:]))
```
Comment (hslatman) [inline ca/ca.go:306]:
```suggestion
			log.Printf("X.509 Root Fingerprint: %s", x509util.Fingerprint(crt))
```
Needs an import of "go.step.sm/crypto/x509util"

**Comment by hslatman** [comment]
Starting to look good 😄 

You can indeed get all addresses the CA is available on using the configured DNS names. Despite the name, it also takes IPs, but internally they're all represented as strings anyway. You can get them from `a.config.DNSNames`, after they've been parsed into that struct from `ca.json`.

**Comment by tashian** [comment]
Ok @hslatman I'd love a final review on this one.

File: commands/app.go [hunk 4]
```diff
@@ -58,6 +58,10 @@ certificate issuer private key used in the RA mode.`,
 			Usage:  "token used to enable the linked ca.",
 			EnvVar: "STEP_CA_TOKEN",
 		},
+		cli.BoolFlag{
+			Name:  "quiet",
+			Usage: "disable startup information",
+		},
```
Comment (hslatman) [inline commands/app.go:64]: It's probably good to also provide the `STEP_CA_QUIET` env var here.

**Comment by hslatman** [comment]
Where's the death metal ASCII art !? 😉

**Comment by tashian** [comment]
Alright, thanks for the notes @hslatman!
I've resolved those.
I also added the version number (how did I miss that?),
and the roots PEM URL.

**Review by hslatman — APPROVED** [review]
lgtm!

**Pull request merged**